lib/charms/mysql/v0/backups.py

```diff
@@ -519,14 +563,41 @@ def _on_restore(self, event: ActionEvent) -> None:
         if not success:
             logger.error(f"Restore failed: {error_message}")
             event.fail(error_message)
-
             if recoverable:
                 self._clean_data_dir_and_start_mysqld()
             else:
+                self.charm.app_peer_data.update({
+                    "s3-block-message": MOVE_RESTORED_CLUSTER_TO_ANOTHER_S3_REPOSITORY_ERROR,
+                    "binlogs-collecting": "",
```

Comment:
I think it's the most convenient way to report a restore error with event fail and blocked unit status simultaneously while keeping `s3-block-message` as application status.
It's important to mention, that while the unit is blocked, application status can't transit to the `s3-block-message` due to the `MySQLOperatorCharm._on_update_status` ->  `if not self.cluster_initialized` -> `if self._mysql.cluster_metadata_exists(self.get_unit_address(unit))`, that will fail like:
```
unit-mysql-0: 01:50:14 WARNING unit.mysql/0.juju-log Failed to check if cluster metadata exists from_instance='10.88.216.67'
unit-mysql-0: 01:50:14 INFO unit.mysql/0.juju-log skip status update when not initialized
```
Summarizing, if this particular check fails, the application status will be next:
<img width="948" alt="image" src="https://github.com/user-attachments/assets/7c972e50-5525-42f6-a4b3-9a86b9cc45b6" />
Also, right below your code reference, there is the next code block:

https://github.com/canonical/mysql-operator/blob/c730506f05f0bdff73d6be6056822933b9b081d4/lib/charms/mysql/v0/backups.py#L578-L581

^ it will imply the same logic I talking about if any of the later backup steps are failed such as `_clean_data_dir_and_start_mysqld`, `_pitr_restore` or `_post_restore`.